src/flows/data_retention/user_snowplow_events_flow.py

```diff
@@ -0,0 +1,31 @@
+"""Delete all raw snowplow events for users who have requested to have their accounts deleted.
+
+ Reference: https://getpocket.atlassian.net/wiki/spaces/CP/pages/2703949851/Snowflake+Data+Deletion+and+Retention
+"""
+
+from prefect import Flow
+from prefect.schedules import Schedule
+from prefect.schedules.clocks import CronClock
+from prefect.tasks.dbt import DbtCloudRunJob
+
+from api_clients.pocket_snowflake_query import PocketSnowflakeQuery
+from utils.flow import get_flow_name
+
+FLOW_NAME = get_flow_name(__file__)
+
+# Run first day of month at midnight UTC.
+SCHEDULE = Schedule(clocks=[CronClock("0 0 1 * *")])
+
+DBT_MODEL_CLEANUP_JOB_ID = 17774
+
+DELETE_SQL = """
+DELETE FROM "SNOWPLOW"."ATOMIC"."EVENTS" as e
+USING "ANALYTICS"."DBT_STAGING"."STG_ACCOUNT_DELETIONS" as d 
+WHERE e.CONTEXTS_COM_POCKET_USER_1[0]:hashed_user_id = d.hashed_user_id
+"""
```

Comment:
I always get nervous about `DELETE` statements on tables like this, and check with a `SELECT` before hand, but alas, the `USING` part doesn't work with a `SELECT`:

```
-- DELETE 
SELECT *
FROM "SNOWPLOW"."ATOMIC"."EVENTS" as e
USING "ANALYTICS"."DBT_STAGING"."STG_ACCOUNT_DELETIONS" as d 
WHERE e.CONTEXTS_COM_POCKET_USER_1[0]:hashed_user_id = d.hashed_user_id
AND d.happened_at > DATEADD("day", -90, NOW())
```
This is simple and looks good to me, but would love a second 👀 + ✅ from @kirill-demtchouk , so we don't have to test out the limits of Snowflake's [Time Travel](https://docs.snowflake.com/en/user-guide/data-availability.html) feature for the first time.

Reply:
If we want to be extra-sure we're deleting all user data, I think we may want to scan a few more entities/fields here besides `hashed_user_id`. These are all of the entities including hashed/de-coded user ID.

Some of these may be redundant, but I figure we want to be thorough. I also added a type cast to `hashed_user_id`:
```
WHERE (e.CONTEXTS_COM_POCKET_USER_1[0]:hashed_user_id::string = d.hashed_user_id
  or e.CONTEXTS_COM_POCKET_USER_1[0]:user_id::integer = d.user_id
  or e.CONTEXTS_COM_POCKET_ACCOUNT_1[0]:hashed_user_id::string = d.hashed_user_id
  or e.CONTEXTS_COM_POCKET_ACCOUNT_1[0]:user_id::integer = d.user_id
  or e.CONTEXTS_COM_POCKET_PAYMENT_SUBSCRIPTION_1[0]:user_id::integer = d.user_id
)

Reply:
I'm not against this, and appreciate being thorough. One thing that @jeshuaborges and I discussed yesterday was that if `hashed_user_id` is not available in event but it is related to a user account, that is either an incomplete/incorrect implementation of the event, or an implementation bug. And that we should fix the event, not account for the incorrect implementation in this process. That makes sense to me from a cleanliness/responsibility standpoint, but in order to go that route I think we would want a way to either enforce that with code or some sort of alerting/monitoring thing. I'm personally not sure what to do in this case, for our H1 milestone.

Reply:
In a casual scan of our dbt models, I think the vast majority have `hashed_user_id` filled-in via the `user` entity. I was being conservative with that above suggestion haha. I agree that we could probably get away with filtering on just `hashed_user_id` in the `user` entity.

Reply:
Lots in this thread and a lot of good comments.

> Re the query. 

100% we should be very confident that our query is doing exactly what we want before we deploy this. I would expect that once we give the thumbs up for this approach we do the select then delete approach as outlined above, in a console. If all looks good then we should move forward with taking this to prod.

> Re the possible missed context and larger where statement:

+1 to what @kennylong said. I really like the idea of having very specific requirements for our clients. If the clients are not meeting the requirement let's sort it out with the client. What I don't want to do is to keep trying to find the place that user ids may be in every event or entity remember to update our deletion processes when we make changes. If it turns out that it is unrealistic for clients to fulfill this expectation then we can follow up with another approach for standardizing the deletes.